common/src/test/java/com/google/auto/common/MoreTypesIsTypeOfTest.java

```diff
@@ -15,198 +15,171 @@
-  }
-
-  private interface PrimitiveVoid {
-    void method();
-  }
+    assertWithMessage("Mirror:\t" + type + "\nClass:\t" + Integer[].class.getCanonicalName())
+        .that(MoreTypes.isTypeOf(Integer[].class, type))
+        .isFalse();
+    assertWithMessage("Mirror:\t" + type + "\nClass:\t" + int[].class.getCanonicalName())
+        .that(MoreTypes.isTypeOf(int[].class, type))
+        .isFalse();
 
-  @Test
-  public void isTypeOf_primitiveVoid() {
-    assertTrue(MoreTypes.isType(typeElementFor(PrimitiveVoid.class).asType()));
-    TypeMirror primitive = extractReturnTypeFromHolder(typeElementFor(PrimitiveVoid.class));
-    assertWithMessage("mirror of a void").that(MoreTypes.isTypeOf(Void.TYPE, primitive)).isTrue();
+    type = typeUtils.getArrayType(typeUtils.getPrimitiveType(TypeKind.INT));
+    assertWithMessage("Mirror:\t" + type + "\nClass:\t" + String[].class.getCanonicalName())
+        .that(MoreTypes.isTypeOf(String[].class, type))
+        .isFalse();
+    assertWithMessage("Mirror:\t" + type + "\nClass:\t" + Integer[].class.getCanonicalName())
+        .that(MoreTypes.isTypeOf(Integer[].class, type))
+        .isFalse();
+    assertWithMessage("Mirror:\t" + type + "\nClass:\t" + int[].class.getCanonicalName())
+        .that(MoreTypes.isTypeOf(int[].class, type))
+        .isTrue();
   }
 
-  private interface DeclaredVoid {
-    Void method();
+  private interface TestType {
+    @SuppressWarnings("unused")
+    <T extends SortedMap<Number, String>> T method0();
   }
 
   @Test
-  public void isTypeOf_declaredVoid() {
-    assertTrue(MoreTypes.isType(typeElementFor(DeclaredVoid.class).asType()));
-    TypeMirror declared = extractReturnTypeFromHolder(typeElementFor(DeclaredVoid.class));
-    assertWithMessage("mirror of a void").that(MoreTypes.isTypeOf(Void.class, declared)).isTrue();
+  public void isTypeOf_declaredType() {
+    TypeMirror TestTypeTypeMirror = getTypeElementFor(TestType.class).asType();
+    assertTrue(MoreTypes.isType(TestTypeTypeMirror));
+    assertWithMessage(
+            "Mirror:\t" + TestTypeTypeMirror + "\nClass:\t" + TestType.class.getCanonicalName())
+        .that(MoreTypes.isTypeOf(TestType.class, TestTypeTypeMirror))
+        .isTrue();
+    assertWithMessage(
+            "Mirror:\t" + TestTypeTypeMirror + "\nClass:\t" + String.class.getCanonicalName())
+        .that(MoreTypes.isTypeOf(String.class, TestTypeTypeMirror))
+        .isFalse();
   }
 
   @Test
   public void isTypeOf_fail() {
     assertFalse(
-        MoreTypes.isType(
-            getOnlyElement(typeElementFor(DeclaredVoid.class).getEnclosedElements()).asType()));
-    TypeMirror method =
-        getOnlyElement(typeElementFor(DeclaredVoid.class).getEnclosedElements()).asType();
+        MoreTypes.isType(getTypeElementFor(TestType.class).getEnclosedElements().get(0).asType()));
+    TypeMirror methodType = getTypeElementFor(TestType.class).getEnclosedElements().get(0).asType();
```

Comment:
I think you could move this before the assertion and use it there? Also, `Iterables.getOnlyElement` might be a bit better than `.get(0)`.